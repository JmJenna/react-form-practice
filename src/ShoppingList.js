import React , {useState} from 'react';
import{v4 as uuid} from 'uuid';
import Item from './Item';
import NewItemForm from './NewItemForm';

const ShoppingList = ()=>{
    const INITIAL_STATE = [
        {id : uuid() , name : "Peanut Butter" , qty : 2},
        {id : uuid() , name : "Oat Milk" , qty : 1},
    ]
    const  [items , setItems ] = useState(INITIAL_STATE)
    const addItem = (newItems)=>{
        setItems(items=>[...items , {...newItems, id: uuid()}])
    }
    return (
        <div>
            <h3>Shopping List</h3>
            <NewItemForm addItem={addItem}/>
            <div>
                 {items.map(({id ,name , qty})=><Item name={name} qty={qty} key={id}/>)}
            </div>
        </div>
    )
}

export default ShoppingList;