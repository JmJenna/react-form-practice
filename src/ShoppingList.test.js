import React from 'react';
import {render , fireEvent} from '@testing-library/react';
import ShoppingList from './ShoppingList';
import { EXPECTED_COLOR } from 'jest-matcher-utils';

it("renders withut crashing", function(){
    render(<ShoppingList />);
});


it("matches snapshot" , function(){
    const { asFragment } = render(<ShoppingList />);
    expect(asFragment()).toMatchSnapshot();
})

it("should add new item", function(){
    const {queryByText} = render(<ShoppingList />);
    const input = queryByLabelText("Product")
    const btn = queryByText("add Item")
    expect(queryByText("Product name : Chocolate Milk")).not.toBeInTheDocument();
    fireEvent.change(input ,{traget: {value : 'Chocolate Milk'}});
    fireEvent.click(btn);
    expect(queryByText("Product name : Chocolate Milk")).toBeInTheDocument();
})



