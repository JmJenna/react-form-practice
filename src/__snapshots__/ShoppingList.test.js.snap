// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Shopping List
    </h3>
    <form>
      <label
        for="name"
      >
        Product :
      </label>
      <input
        id="name"
        name="name"
        placeholder="Item name"
        type="text"
        value=""
      />
      <label
        for="qty"
      >
        Quantity :
      </label>
      <input
        id="qty"
        name="qty"
        placeholder="Quantity"
        type="text"
        value=""
      />
      <button>
        Add Item
      </button>
    </form>
    <div>
      <ul>
        <li>
          Product name : Peanut Butter
        </li>
        <li>
          Qty : 2 
        </li>
      </ul>
      <ul>
        <li>
          Product name : Oat Milk
        </li>
        <li>
          Qty : 1 
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
